app/recipes/tests/test_recipes_api.py

```diff
@@ -0,0 +1,259 @@
+from django.urls import reverse
+from django.test import TestCase
+from rest_framework import status
+from rest_framework.test import APIClient
+
+from core.models import Recipe, Ingredient
+
+# Recipes collection resource URL, i.e. /recipes/
+RECIPES_URL = reverse('recipes:recipe-list')
+
+
+def recipe_url(recipe_id):
+    """Return recipe resource URL, i.e. /recipes/{recipe_id}/"""
+    return reverse('recipes:recipe-detail', args=[recipe_id])
+
+
+def create_sample_recipe(name, ingredient_names, description=''):
+    recipe = Recipe.objects.create(
+        name=name,
+        description=description,
+    )
+    for ingredient_name in ingredient_names:
+        Ingredient.objects.create(
+            name=ingredient_name,
+            recipe=recipe
+        )
+    return recipe
+
+
+class RecipeApiTests(TestCase):
+    """Test suite for recipes API endpoints"""
+
+    def setUp(self):
+        self.client = APIClient()
+
+    def test_create_recipe_with_description(self):
+        """POST /recipes/ should create recipe and return new resource
+            representation, including ID"""
+        payload = {
+            'name': 'Lemon cheesecake',
+            'description': 'Cheesecake (feat. lemon)',
+            'ingredients': [
+                {'name': 'Lemon'},
+                {'name': 'Cheese'}
+            ]
+        }
+
+        response = self.client.post(RECIPES_URL, payload, format='json')
+
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        self.assertEqual(response.json(), {
+            'id': response.data['id'],
+            'name': 'Lemon cheesecake',
```

Comment:
You could use destructuring here, `{'id': response.data['id'], **payload}` IMO this would make the test a bit more readable, making it obvious that the payload is being repeated back.

Reply:
Good to know - thanks! Somehow I'd got the idea that `**` only worked for passing named function params